File: evals/fixtures/hexagonal/typescript/src/adapter.ts
import type {
  PaymentAuthorizer,
  PaymentReceipt,
  PaymentRequest,
} from "./application";

export type VendorCharge = Readonly<{ providerId: string }>;

export interface VendorClient {
  charge(amount: number, requestKey: string): Promise<VendorCharge>;
}

export class VendorPaymentAdapter implements PaymentAuthorizer {
  constructor(private readonly client: VendorClient) {}

  async authorize(request: PaymentRequest): Promise<PaymentReceipt> {
    const charge = await this.client.charge(
      request.amountMinor,
      request.idempotencyKey,
    );
    return { paymentId: charge.providerId };
  }
}

File: evals/fixtures/hexagonal/python/hex_fixture/adapter.py
from dataclasses import dataclass
from typing import Protocol

from hex_fixture.application import PaymentReceipt, PaymentRequest


@dataclass(frozen=True)
class VendorCharge:
    provider_id: str


class VendorClient(Protocol):
    def charge(self, amount: int, request_key: str) -> VendorCharge: ...


class VendorPaymentAdapter:
    def __init__(self, client: VendorClient) -> None:
        self._client = client

    def authorize(self, request: PaymentRequest) -> PaymentReceipt:
        charge = self._client.charge(request.amount_minor, request.idempotency_key)
        return PaymentReceipt(charge.provider_id)
